core/src/main/scala/org/ensime/core/Project.scala

```diff
@@ -69,6 +72,9 @@ class Project(
         // we could also just blindly send this on each connection.
         broadcaster ! Broadcaster.Persist(IndexerReadyEvent)
         log.debug(s"created $inserts and removed $deletes searchable rows")
+        val exitAfterIndex = propOrFalse("ensime.exitAfterIndex")
```

Comment:
A more idomatic was to do this would be to listen to the broadcaster event and triggering the shutdown - this code is effectively  a weird side effect on the behaviour here.